<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>614</width>
    <height>340</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MakeWaves</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabwidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>271</width>
      <height>231</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Regular</string>
     </attribute>
     <widget class="QComboBox" name="combobox_regparams">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>141</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Height and Period</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Height and Wavelength</string>
       </property>
      </item>
     </widget>
     <widget class="QDoubleSpinBox" name="spinbox_wave_height">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>50</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="accelerated">
       <bool>true</bool>
      </property>
      <property name="decimals">
       <number>3</number>
      </property>
      <property name="maximum">
       <double>0.400000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="spinbox_wave_period">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>80</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="accelerated">
       <bool>true</bool>
      </property>
      <property name="decimals">
       <number>3</number>
      </property>
      <property name="minimum">
       <double>0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>5.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="spinbox_wavelength">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>110</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="accelerated">
       <bool>true</bool>
      </property>
      <property name="decimals">
       <number>3</number>
      </property>
      <property name="singleStep">
       <double>0.001000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Wave Height (m):</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Wave Period (s):</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Wavelength (m):</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QwtSlider" name="slider_horiz">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
      <property name="valid">
       <bool>true</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="scalePosition">
       <enum>QwtSlider::BottomScale</enum>
      </property>
      <property name="bgStyle">
       <enum>QwtSlider::BgSlot</enum>
      </property>
      <property name="thumbLength">
       <number>16</number>
      </property>
      <property name="thumbWidth">
       <number>12</number>
      </property>
      <property name="borderWidth">
       <number>2</number>
      </property>
     </widget>
     <widget class="QwtSlider" name="slider_height">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>40</y>
        <width>61</width>
        <height>141</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="scalePosition">
       <enum>QwtSlider::RightScale</enum>
      </property>
      <property name="bgStyle">
       <enum>QwtSlider::BgSlot</enum>
      </property>
      <property name="thumbLength">
       <number>16</number>
      </property>
      <property name="thumbWidth">
       <number>12</number>
      </property>
      <property name="borderWidth">
       <number>2</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Parameters:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Random</string>
     </attribute>
     <widget class="QComboBox" name="combobox_randwavetype">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>111</width>
        <height>22</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Bretschneider</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>JONSWAP</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>NH Extreme</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>NH Typical</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Pierson Moscowitz</string>
       </property>
      </item>
     </widget>
     <widget class="QTableWidget" name="table_rwaves">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>241</width>
        <height>161</height>
       </rect>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="showDropIndicator" stdset="0">
       <bool>false</bool>
      </property>
      <property name="dragDropOverwriteMode">
       <bool>false</bool>
      </property>
      <property name="alternatingRowColors">
       <bool>false</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::NoSelection</enum>
      </property>
      <property name="showGrid">
       <bool>true</bool>
      </property>
      <property name="cornerButtonEnabled">
       <bool>false</bool>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>175</number>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>30</number>
      </attribute>
      <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderCascadingSectionResizes">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderDefaultSectionSize">
       <number>20</number>
      </attribute>
      <attribute name="verticalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderMinimumSectionSize">
       <number>15</number>
      </attribute>
      <attribute name="verticalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <row>
       <property name="text">
        <string>0</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>1</string>
       </property>
      </row>
      <row>
       <property name="text">
        <string>2</string>
       </property>
      </row>
      <column>
       <property name="text">
        <string>Parameter</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Value</string>
       </property>
      </column>
      <item row="0" column="0">
       <property name="text">
        <string>Significant Wave Height (m)</string>
       </property>
      </item>
      <item row="0" column="1">
       <property name="text">
        <string>0.1</string>
       </property>
       <property name="flags">
        <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
       </property>
      </item>
      <item row="1" column="0">
       <property name="text">
        <string>Significant Wave Period (s)</string>
       </property>
      </item>
      <item row="1" column="1">
       <property name="text">
        <string>1</string>
       </property>
       <property name="flags">
        <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
       </property>
      </item>
      <item row="2" column="0">
       <property name="text">
        <string>Scale Ratio</string>
       </property>
      </item>
      <item row="2" column="1">
       <property name="text">
        <string>1</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Spectrum Type:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QDockWidget" name="dock_spectrum">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>300</y>
      <width>325</width>
      <height>256</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>325</width>
      <height>256</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>325</width>
      <height>256</height>
     </size>
    </property>
    <property name="floating">
     <bool>false</bool>
    </property>
    <property name="windowTitle">
     <string>Output Spectrum</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents">
     <widget class="QwtPlot" name="plot_spec">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>291</width>
        <height>201</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 8pt &quot;MS Shell Dlg 2&quot;;</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QDockWidget" name="dock_time_series">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>580</y>
      <width>325</width>
      <height>256</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>325</width>
      <height>256</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>325</width>
      <height>256</height>
     </size>
    </property>
    <property name="windowTitle">
     <string>Output Time Series</string>
    </property>
    <widget class="QWidget" name="dockWidgetContents_2">
     <widget class="QwtPlot" name="plot_ts">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>291</width>
        <height>201</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 8pt &quot;MS Shell Dlg 2&quot;;</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>614</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="action_view_ts"/>
    <addaction name="action_view_spec"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="action_wiki"/>
    <addaction name="separator"/>
    <addaction name="action_about"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="action_editsettings"/>
    <addaction name="actionRecording"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolbar_main">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>Main Toolbar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_start"/>
   <addaction name="action_record"/>
  </widget>
  <widget class="QDockWidget" name="dock_measure">
   <property name="minimumSize">
    <size>
     <width>325</width>
     <height>265</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Measured Values</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <widget class="QwtPlot" name="plot_meas">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>291</width>
       <height>201</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 8pt &quot;MS Shell Dlg 2&quot;;</string>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="actionRegular">
   <property name="text">
    <string>Regular</string>
   </property>
  </action>
  <action name="actionRandom">
   <property name="text">
    <string>Random</string>
   </property>
  </action>
  <action name="action_view_ts">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Output Time Series</string>
   </property>
  </action>
  <action name="action_view_spec">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Output Spectrum</string>
   </property>
  </action>
  <action name="action_wiki">
   <property name="text">
    <string>Tow Tank Wiki</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="action_editsettings">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Recalculate Limits</string>
   </property>
  </action>
  <action name="action_start">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/play.png</normaloff>:/icons/play.png</iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
   <property name="toolTip">
    <string>Start Generating Waves</string>
   </property>
  </action>
  <action name="action_record">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/record.png</normaloff>:/icons/record.png</iconset>
   </property>
   <property name="text">
    <string>Record</string>
   </property>
   <property name="toolTip">
    <string>Enable Data Recording</string>
   </property>
  </action>
  <action name="actionRecording">
   <property name="text">
    <string>Recording...</string>
   </property>
  </action>
  <zorder>dock_measure</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
  </customwidget>
  <customwidget>
   <class>QwtSlider</class>
   <extends>QWidget</extends>
   <header>qwt_slider.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>264</x>
     <y>277</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_view_ts</sender>
   <signal>triggered(bool)</signal>
   <receiver>dock_time_series</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dock_time_series</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>action_view_ts</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>435</x>
     <y>438</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_view_spec</sender>
   <signal>triggered(bool)</signal>
   <receiver>dock_spectrum</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dock_spectrum</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>action_view_spec</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>435</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
